app/assets/javascripts/discourse/controllers/pages_controller.js

```diff
@@ -0,0 +1,51 @@
+/**
+  This controller supports displaying custom pages.
+
+  @class PagesController
+  @extends Discourse.Controller
+  @namespace Discourse
+  @module Discourse
+**/
+Discourse.PagesController = Discourse.Controller.extend({
+  content: null,
+
+  loadPage: function(path) {
+    var $preloaded, text,
+      _this = this;
+    this.set('content', null);
+    // Load from <noscript> if we have it.
+    $preloaded = $("noscript[data-path=\"" + path + "\"]");
```

Comment:
I remember this code from somewhere else. Can't it be extracted to one utilitary class/method and be used by both place?